server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -266,7 +267,8 @@ protected final DocsStats docsStats(IndexReader indexReader) {
      */
     public ShardFieldStats shardFieldStats() {
         try (var searcher = acquireSearcher("shard_field_stats", Engine.SearcherScope.INTERNAL)) {
-            return shardFieldStats(searcher.getLeafContexts());
+            boolean isStateless = DiscoveryNode.isStateless(engineConfig.getIndexSettings().getNodeSettings());
```

Comment:
Can we turn this into a field in `Engine`? Then there is no need to read the node settings every time this method gets invoked.